packages/core/core-flows/src/order/workflows/create-shipment.ts

```diff
@@ -0,0 +1,140 @@
+import { FulfillmentDTO, OrderDTO, OrderWorkflow } from "@medusajs/types"
+import {
+  MedusaError,
+  Modules,
+  OrderStatus,
+  arrayDifference,
+} from "@medusajs/utils"
+import {
+  WorkflowData,
+  createStep,
+  createWorkflow,
+  transform,
+} from "@medusajs/workflows-sdk"
+import { useRemoteQueryStep } from "../../common"
+import { createShipmentWorkflow } from "../../fulfillment"
+import { registerOrderShipmentStep } from "../steps"
+
+function throwIfOrderIsCancelled({ order }: { order: OrderDTO }) {
+  if (order.status === OrderStatus.CANCELED) {
+    throw new MedusaError(
+      MedusaError.Types.INVALID_DATA,
+      `Order with id ${order.id} has been cancelled.`
+    )
+  }
+}
+
+function throwIfItemsDoesNotExistsInOrder({
+  order,
+  inputItems,
+}: {
+  order: Pick<OrderDTO, "id" | "items">
+  inputItems: OrderWorkflow.CreateOrderShipmentWorkflowInput["items"]
+}) {
+  const orderItemIds = order.items?.map((i) => i.id) ?? []
+  const inputItemIds = inputItems.map((i) => i.id)
+  const diff = arrayDifference(inputItemIds, orderItemIds)
+
+  if (diff.length) {
+    throw new MedusaError(
+      MedusaError.Types.INVALID_DATA,
+      `Items with ids ${diff.join(", ")} does not exist in order with id ${
+        order.id
+      }.`
+    )
+  }
+}
```

Comment:
suggestion: what do you think if we move some validations in the order utils dir next to the workflow? I believe many workflows will needs those validations right?

Reply:
good call.